src/Servers/Kestrel/shared/KnownHeaders.cs

```diff
@@ -213,8 +112,6 @@ public static string GeneratedFile()
                 PrimaryHeader = requestPrimaryHeaders.Contains("Content-Length")
             }})
             .ToArray();
-            Debug.Assert(requestHeaders.Length <= 64);
-            Debug.Assert(requestHeaders.Max(x => x.Index) <= 62);
```

Comment:
Why did these Debug.Asserts get removed? It looks like we're still using `KnownHeader.TestBit()`, so I think it's still a good check to have.

Reply:
Moved it to the start of the `GeneratedFile()` method line 527; also added check for responseHeaders.